consensus/mempool_test.go

```diff
@@ -25,25 +25,31 @@ func assertMempool(txn txNotifier) mempl.Mempool {
 }
 
 func TestMempoolNoProgressUntilTxsAvailable(t *testing.T) {
-	config := ResetConfig("consensus_mempool_txs_available_test")
-	defer os.RemoveAll(config.RootDir)
-	config.Consensus.CreateEmptyBlocks = false
-
-	state, privVals := randGenesisState(1, false, 100)
-
-	cs := newStateWithConfig(config, state, privVals[0], NewCounterApplication())
-	assertMempool(cs.txNotifier).EnableTxsAvailable()
-	height, round := cs.Height, cs.Round
-	newBlockCh := subscribe(cs.eventBus, types.EventQueryNewBlock)
-	startTestRound(cs, height, round)
-
-	ensureNewEventOnChannel(newBlockCh) // first block gets committed
-	ensureNoNewEventOnChannel(newBlockCh)
-	deliverTxsRange(cs, 0, 1)
-	ensureNewEventOnChannel(newBlockCh) // commit txs
-	ensureNewEventOnChannel(newBlockCh) // commit updated app hash
-	ensureNewEventOnChannel(newBlockCh) // commit 2nd block for updated app hash
-	ensureNoNewEventOnChannel(newBlockCh)
+	for proofBlockRange := int64(1); proofBlockRange <= 3; proofBlockRange++ {
+		t.Logf("Checking proof block range %d", proofBlockRange)
+
+		config := ResetConfig("consensus_mempool_txs_available_test")
```

Comment:
can we move the creation of the config out of the loop?

Reply:
It's not trivial, and each test run should be independent. If I move this initialization code, it will create the configs, data directory etc. and each subsequent run will reuse it.